packages/core/src/transport/batch.ts

```diff
@@ -70,12 +78,12 @@ export class Batch {
       this.remove(key)
     }
     if (this.willReachedBytesLimitWith(messageBytesCount)) {
-      this.flush()
+      this.flush('batch_bytes_limit')
     }
 
     this.push(processedMessage, messageBytesCount, key)
     if (this.isFull()) {
-      this.flush()
+      this.flush('batch_bytes_limit')
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏We could have two flush reason, one for the bytes limit and the other for messages limit

Reply:
True, however for the scope of this investigation (assess pagehide impact) I think it's okay. Wdyt?

Reply:
Alright, I'm fine with that if it's temporary